### PR TITLE
port export_gen, export_cassandra, export_solr

### PR DESCRIPTION
**Pull request by jbloom22**
Includes changes from porting export_vcf #2728

### CODE REVIEW

**Comment by danking** [comment]
@jbloom22 this PR is unassigned

**Comment by jbloom22** [comment]
yes, I was waiting for Tim to approve my first port PR so I could make any formatting changes there to this one as well. The former just went in so I'm updating this now and will then assign.

File: python/hail/methods/io.py [hunk 1]
```diff
@@ -1,18 +1,114 @@
...
+           keyspace=strlike,
+           table_name=strlike,
+           block_size=integral,
+           rate=integral)
+def export_cassandra(table, address, keyspace, table_name, block_size=100, rate=1000):
+    """Export to Cassandra.
+
+    Warning
+    -------
+    :meth:`export_cassandra` is EXPERIMENTAL.
```
Comment (patrick-schultz) [inline python/hail/methods/io.py]: I think this should be `` :func:`.export_cassandra` ``. That's the proper semantic markup, and it looks like with `:meth:` sphinx doesn't find the reference, and doesn't add the link in the html.

See http://www.sphinx-doc.org/en/stable/domains.html#python-roles

File: python/hail/methods/io.py [hunk 2]
```diff
@@ -1,18 +1,114 @@
...
+    - ID_1 and ID_2 are identical and set to the sample ID (`s`).
+    - The third column (``missing``) is set to 0 for all samples.
+
+    Parameters
+    ----------
+    dataset : :class:`.MatrixTable`
+        Dataset with entry field `GP` of type Array[TFloat64].
+    output : :obj:`str`
+        Output file root for GEN and SAMPLE files.
+    precision : int
```
Comment (patrick-schultz) [inline python/hail/methods/io.py]: It looks odd to me having `str` in the `:obj:` markup but not int. But I don't think we've agreed on a style.

File: python/hail/methods/io.py [hunk 3]
```diff
@@ -1,18 +1,114 @@
...
+@typecheck(table=Table,
+           zk_host=strlike,
+           collection=strlike,
+           block_size=integral)
+def export_solr(table, zk_host, collection, block_size=100):
+    """Export to Solr.
+    
+    Warning
+    -------
+    :meth:`export_solr` is EXPERIMENTAL.
```
Comment (patrick-schultz) [inline python/hail/methods/io.py]: Same as above, should be `` :meth:`.export_solr` ``.

**Comment by danking** [comment]
@jbloom22 needs a rebase

**Pull request merged**